app/models/miq_provision/state_machine.rb

```diff
@@ -43,7 +43,10 @@ def poll_destination_powered_on_in_provider
 
   def poll_destination_in_vmdb
     update_and_notify_parent(:message => "Validating New #{destination_type}")
-
+    # fill vm_name field if is not given
+    if dest_name && phase_context[:new_vm_ems_ref] && phase_context[:new_vm_ems_ref][:vm_name].nil?
+      phase_context[:new_vm_ems_ref][:vm_name] = dest_name
+    end
```

Comment:
Use `fetch_path` and `store_path` and drop comment line:

```ruby
if dest_name && phase_context.fetch_path(:new_vm_ems_ref, :vm_name).nil?
  phase_context.store_path(:new_vm_ems_ref, :vm_name, dest_name)
end
```

Also needs tests